src/forwarder/internal/pkg/cmd/run.go

```diff
@@ -0,0 +1,148 @@
+	"github.com/spf13/cobra"
+	"google.golang.org/grpc"
+
+	"github.com/networkservicemesh/sdk/pkg/tools/flags"
+	"github.com/networkservicemesh/sdk/pkg/tools/grpcoptions"
+
+	"github.com/networkservicemesh/cmd-forwarder-vppagent/src/forwarder/internal/pkg/vppinit"
+)
+
+func init() {
+	cmd := runCmd
+	rootCmd.AddCommand(cmd)
+	Flags(cmd.Flags())
+	cobra.OnInitialize(flags.FromEnv(flags.EnvPrefix, flags.EnvReplacer,cmd.Flags()))
+}
+
+var runCmd = &cobra.Command{
+	Use:   "run",
+	Short: "Runs xconnect network service",
+	Long: `Runs xconnect network service.  Supported mechanisms:
+     - memif
+     - kernel
+     - vxlan`,
+	Run: func(cmd *cobra.Command, args []string) {
+		// Capture signals to cleanup before exiting - note: this *must* be the first thing in main
+		c := make(chan os.Signal, 1)
+		signal.Notify(c,
+			os.Interrupt,
+			// More Linux signals here
+			syscall.SIGHUP,
+			syscall.SIGTERM,
+			syscall.SIGQUIT)
+
+		starttime := time.Now()
+		// Setup logging
+		logrus.SetReportCaller(true)
+		// TODO - decide whether or not to continue to use nested formatter
+		// nested formatter does *not* escape \n, and so backtraces in logs are usable
+		// during dev, but its probably undesirable for general use
+		logrus.SetFormatter(&nested.Formatter{})
+
+		// Context to use for all things started in main
+		ctx, cancelFunc := context.WithCancel(context.Background())
+		defer cancelFunc()
+
+		// grpc.ClientConn for dialing the vppagent
+		vppagentCC, err := grpc.DialContext(ctx, VppAgentURL.String(), grpc.WithInsecure())
+		if err != nil {
+			logrus.Fatalf("failed to dial vppagent with %+v", err)
+		}
+
+		// OpenPolicyAgent Authz Policy
+		policyBytes,err := ioutil.ReadFile(AuthzPolicyFile)
+		if err != nil {
+			logrus.Fatalf("Unable to open Authz policy file %q",AuthzPolicyFile)
+		}
+		policy := string(policyBytes)
+		authzPolicy, err := rego.New(
+			// TODO - rework the example.com and test pieces here
+			rego.Query("data.test.allow"),
+			rego.Module("example.com", policy)).PrepareForEval(context.Background())
+
+		// XConnect Network Service Endpoint
+		endpoint := xconnectns.NewServer(
+			Name,
+			&authzPolicy,
+			vppagentCC,
+			BaseDir,
+			TunnelIP,
+			vppinit.Func(TunnelIP),
+			&ConnectToURL,
+			grpcoptions.WithSpiffe(&SpiffeAgentURL, 10*time.Millisecond),
+		)
+
+		// Create listener
+		network, target := UrlToNetworkTarget(&ListenOnURL)
+		ln, err := net.Listen(network, target)
+		defer ln.Close()
+		if err != nil {
+			logrus.Fatalf("failed to listen on %q: %v", ListenOnURL,err)
+		}
+
+		// Create GRPC Server
+		// TODO - add ServerOptions for Tracing
+		server := grpc.NewServer(grpcoptions.SpiffeCreds(&SpiffeAgentURL, 10*time.Millisecond))
+		endpoint.Register(server)
+
+		// Serve
+		go func() {
+			err = server.Serve(ln)
+			if err != nil {
+				logrus.Fatalf("failed to serve on %q: %v", ListenOnURL,err)
+			}
+		}()
+		logrus.Infof("Startup completed in %v", time.Since(starttime))
```

Comment:
I've not found a logic to register NSE with NSMgr.

Reply:
You are correct, not yet, should be easy enough to add, trying to get something working at a bit of a higher level here first.